src/WebAPI/Models/RecommendationRepository.cs

```diff
@@ -26,6 +26,16 @@ public RecommendationRepository(IOptions<DbConnectionConfig> dbConnectionConfig)
         }
     }
 
+    public async Task<IEnumerable<Restaurant>> GetRestaurantsUsingFilter(int pageOffset, int pageLimit)
+    {
+        try {
+            return await _context.Recommendations.Find(_ => true).Skip(pageOffset*pageLimit).Limit(pageLimit).ToListAsync();
```

Comment:
Can you return an object that has a property for total rows, and a property for total pages as well?